tests/RobotFramework/tests/mqtt/remote_mqtt_broker.robot

```diff
@@ -0,0 +1,93 @@
+*** Settings ***
+Resource    ../../resources/common.resource
+Library    ThinEdgeIO
+Library    Cumulocity
+
+Test Tags    theme:mqtt    theme:c8y    adapter:docker
+Suite Setup    Custom Setup
+Suite Teardown    Get Logs    name=${MAIN_DEVICE}
+
+*** Variables ***
+${MAIN_DEVICE}
+${SUB_COMPONENT}
+
+*** Test Cases ***
+Check remote mqtt broker #1773
+    [Documentation]    The test relies on two containers functioning as one logical device. 1 container (the MAIN_DEVICE)
+    ...    runs the mqtt broker and the mapper, and the other container (SUB_COMPONENT) runs the other tedge components.
+    ...    Together the two containers should function similar to installing everything in one container (more or less).
+    ...    This is the building blocks for enabling a single-process container setup where each thin-edge.io component
+    ...    is running in its own container.
+    [Tags]    \#1773
+    ThinEdgeIO.Set Device Context    ${MAIN_DEVICE}
+    ThinEdgeIO.Service Should Be Running    mosquitto
+    ThinEdgeIO.Service Should Be Running    tedge-mapper-c8y
+    ThinEdgeIO.Service Should Be Stopped    tedge-agent
+    ThinEdgeIO.Service Should Be Stopped    c8y-configuration-plugin
+    ThinEdgeIO.Service Should Be Stopped    c8y-firmware-plugin
+    ThinEdgeIO.Service Should Be Stopped    c8y-log-plugin
+
+    ThinEdgeIO.Set Device Context    ${SUB_COMPONENT}
+    ThinEdgeIO.Service Should Be Stopped    mosquitto
+    ThinEdgeIO.Service Should Be Stopped    tedge-mapper-c8y
+    ThinEdgeIO.Service Should Be Running    tedge-agent
+    ThinEdgeIO.Service Should Be Running    c8y-configuration-plugin
+    ThinEdgeIO.Service Should Be Running    c8y-firmware-plugin
+    ThinEdgeIO.Service Should Be Running    c8y-log-plugin
+
+    # Validate the device exists in the cloud
+    Cumulocity.Device Should Exist    ${MAIN_DEVICE}
+
+    # Cumulocity sanity check
+    ThinEdgeIO.Set Device Context    ${SUB_COMPONENT}
+    ThinEdgeIO.Execute Command    tedge mqtt pub tedge/measurements '{"temperature": 29.8}'
+    ${measurements}=    Cumulocity.Device Should Have Measurements    value=temperature    series=temperature
+    Should Be Equal As Numbers    ${measurements[0]["temperature"]["temperature"]["value"]}    ${29.8}
+
+    Cumulocity.Should Have Services    name=tedge-mapper-c8y    status=up
+    Cumulocity.Should Have Services    name=tedge-agent    status=up
+    Cumulocity.Should Have Services    name=c8y-configuration-plugin    status=up
+    Cumulocity.Should Have Services    name=c8y-firmware-plugin    status=up
+    Cumulocity.Should Have Services    name=c8y-log-plugin    status=up
+
+*** Keywords ***
+
+Custom Setup
+    # Device 1 (mqtt host and mapper)
```

Comment:
Above in the documentation, it's mentioned `MAIN_DEVICE` and `SUB_COMPONENT`. Here it's mentioned as `device 1 & Device 2` I feel we have to be consistent. I feel we can mention `MAIN_DEVICE` & `EXTERNAL_DEVICE`.

Reply:
I wanted to stay away from using "DEVICE" too much as it might lead the user to think that we are talking about multiple device in the cloud. The main purpose of the test is to confirm it will work for containers (and other more exotic setups). I have renamed the variables to use `CONTAINER_1` and `CONTAINER_2` instead.

Reply:
`Conatiner_1 and Container_2` make sense.